nuxeo-mongodb-audit/src/main/java/org/nuxeo/mongodb/audit/MongoDBAuditEntryReader.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc
+ */
+package org.nuxeo.mongodb.audit;
+
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_CATEGORY;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_COMMENT;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_DOC_LIFE_CYCLE;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_DOC_PATH;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_DOC_TYPE;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_DOC_UUID;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_EVENT_DATE;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_EVENT_ID;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_EXTENDED;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_LOG_DATE;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_PRINCIPAL_NAME;
+import static org.nuxeo.mongodb.audit.LogEntryConstants.PROPERTY_REPOSITORY_ID;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.bson.Document;
+import org.nuxeo.ecm.platform.audit.api.ExtendedInfo;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.audit.impl.ExtendedInfoImpl;
+import org.nuxeo.ecm.platform.audit.impl.LogEntryImpl;
+import org.nuxeo.mongodb.core.MongoDBSerializationHelper;
+
+import com.mongodb.DBObject;
+
+/**
+ * Reader for MongoDB Audit.
+ *
+ * @since 9.1
+ */
+public class MongoDBAuditEntryReader {
+
+    public static LogEntry read(Document doc) {
+        LogEntryImpl entry = new LogEntryImpl();
+        for (String key : doc.keySet()) {
+            if (PROPERTY_CATEGORY.equals(key)) {
```

Comment:
Let's use a `switch` instead of this big sequence of `else if`. Also I'd add at least a `log.warn` for the `default:` if we don't find the key in our list.